src/main/groovy/lang/ObjectRange.java

```diff
@@ -37,7 +37,7 @@
  * Note: This class is similar to {@link IntRange}. If you make any changes to this
  * class, you might consider making parallel changes to {@link IntRange}.
  */
-public class ObjectRange extends AbstractList implements Range {
+public class ObjectRange extends AbstractList<Comparable> implements Range<Comparable> {
```

Comment:
ups, did not notice, yes... `public class ObjectRange<T extends Comparable> extends AbstractList<t> implements Range<T> {` was what I was thinking about